libs/core/mappers/anime.mapper.ts

```diff
@@ -0,0 +1,30 @@
+import { AnimeDto } from '../dtos/anime.dto';
+import { Anime, Status, Type } from '../models/anime';
+
+export namespace AnimeMapper {
+
+  /**
+   * Maps dto to model.
+   * @param dto Anime dto.
+   * @param mapperFromDto Callback of result mapper from DTO to Model.
+   */
+  export function fromDto<Dto, Model>(
+    dto: AnimeDto<Dto>,
+    mapperFromDto: (resultDto: Dto) => Model,
+  ): Anime<Model> {
+
+    const status = Object.values(Status).includes(dto.status) ? dto.status : Status.Airing;
+    const type = Object.values(Type).includes(dto.type) ? dto.type : Type.Tv;
+
+    return new Anime({
+      id: dto.id,
+      image: dto.image,
+      titleEnglish: dto.title_eng,
+      titleJapanese: dto.title_jpn,
+      aired: mapperFromDto(dto.aired),
```

Comment:
There is no need to pass mapper here. Just use it directly
```suggestion
      aired: DateRangeMapper.fromDto(dto.aired),
```